types/header.go

```diff
@@ -62,6 +64,28 @@ func (n Nonce) MarshalText() ([]byte, error) {
 	return []byte(n.String()), nil
 }
 
+func (n *Nonce) UnmarshalText(input []byte) error {
+	buf := StringToBytes(string(input))
+	if len(buf) != NonceLength {
+		return fmt.Errorf("incorrect length")
+	}
+
+	*n = BytesToNonce(buf)
+
+	return nil
+}
+
+func BytesToNonce(b []byte) Nonce {
```

Comment:
I feel like this method should belong to Nonce struct.

Something like `n.setBytes(buf)`, wdyt?

Reply:
Totally agree with you, I was just following what was already implemented for `Address` and `Hash`:
- https://github.com/0xPolygon/polygon-edge/blob/109517239e02bb27a3fae5448660e4780bff657a/types/types.go#L125
- https://github.com/0xPolygon/polygon-edge/blob/109517239e02bb27a3fae5448660e4780bff657a/types/types.go#L57

Reply:
In this case no, let's favor consistency.